Sources/Classes/Search/BaseSearchViewModel.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Copyright (c) 2019 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import RxSwift
+import RxCocoa
+
+/// ViewModel that loads data from a given data source and performs search among results
+open class BaseSearchViewModel<Item, ItemViewModel>: GeneralDataLoadingViewModel<[Item]> {
+
+    public typealias ItemsList = [Item]
+
+    private let searchTextRelay = BehaviorRelay(value: "")
+
+    public init(dataSource: Single<ItemsList>) {
+        super.init(dataSource: dataSource, emptyResultChecker: { $0.isEmpty })
+    }
+
+    open var itemsViewModelsDriver: Driver<[ItemViewModel]> {
+        return loadingResultObservable
+            .map { [weak self] items in
+                self?.viewModels(from: items)
+            }
+            .filterNil()
+            .share(replay: 1, scope: .forever)
+            .asDriver(onErrorDriveWith: .empty())
+    }
+
+    open var searchResultsDriver: Driver<[ItemViewModel]> {
+        return searchTextRelay.throttle(1, scheduler: MainScheduler.instance)
```

Comment:
why throttle, not debounce?

Reply:
interval to variable (to be customized), 1 sec is too much, btw

Reply:
fixed